src/features/payment-agreements/hooks/mutations/use-update-payment-agreement-mutation.ts

```diff
@@ -21,6 +21,7 @@ export function useUpdatePaymentAgreementMutation() {
     mutationFn: ({id, input}: UpdateAgreementInput) => PaymentAgreementsService.updateAgreement(id, input),
     onSuccess: async () => {
       await client.resetQueries({queryKey: accountPaymentAgreementsCreatedQueryKey(address), exact: true})
+      await client.resetQueries({queryKey: accountPaymentAgreementById(address, agreementId), exact: true})
       navigate(RoutesConfig.paymentAgreements)
```

Comment:
Navigation (`navigate(...)`) should not happen inside the mutation success handler. What if I want to update something from another page and I don't want to navigate away on success?

The `onSuccess` in mutations should only do query invalidation, nothing else!